src/backends/tensorflow/savedmodel_backend_factory.cc

```diff
@@ -49,6 +50,12 @@ SavedModelBackendFactory::Create(
   auto savedmodel_backend_config =
       std::static_pointer_cast<GraphDefBackendFactory::Config>(backend_config);
   factory->reset(new SavedModelBackendFactory(savedmodel_backend_config));
+
+  // Initialize VGPUs if required
+  if (!savedmodel_backend_config->memory_limit_mb.empty()) {
```

Comment:
You are putting vgpu knowledge outside of VirtualDeviceTracker. You should always call Init and let the tracker determine how to initialize and if any gpu -> vgpu is needed or not.